### PR TITLE
Capsule for HD44780 LCD

### PR DESCRIPTION
**Pull request by teonaseverin**
### Pull Request Overview

This pull request adds a capsule for a HD44780 LCD screen.
This will allow controlling an LCD screen without any delay in userspace. 

### Testing Strategy

This pull request was tested with NucleoF429ZI

### TODO or Help Wanted

Testing on other devices would be great. Any feed-back is appreciated.

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make formatall`.

### CODE REVIEW

**Comment by bradjc** [comment]
Looks good! I just ask that you add it to the list in capsules/README.md.

File: capsules/README.md [hunk 1]
```diff
@@ -93,6 +93,7 @@ These provide common and better abstractions for userspace.
 - **[Button](src/button.rs)**: Detect button presses.
 - **[Buzzer](src/buzzer_driver.rs)**: Simple buzzer.
 - **[Console](src/console.rs)**: UART console support.
+- **[HD44780 LCD](src/hd44780.rs)**: HD44780 LCD screen.
```
Comment (bradjc) [inline capsules/README.md]: Can you move this to line 48? Or if you want you can add another category under "Sensor and other IC Drivers".

File: boards/nucleo_f429zi/Makefile [hunk 2]
```diff
@@ -8,6 +8,10 @@ include ../Makefile.common
 OPENOCD=openocd
 OPENOCD_OPTIONS=-f openocd.cfg
 
+APP=../../../libtock-c/examples/lcd_16x2/build/cortex-m4/cortex-m4.tbf
+KERNEL=$(CWD)/target/$(TARGET)/debug/$(PLATFORM).elf
+KERNEL_WITH_APP=$(CWD)/target/$(TARGET)/debug/$(PLATFORM)-app.elf
+
```
Comment (hudson-ayers) [inline boards/nucleo_f429zi/Makefile]: I don't think we want to merge any changes to the nucleo Makefile, not everyone will have libtock-c in that relative directory, for example.

File: boards/nucleo_f429zi/src/main.rs [hunk 3]
```diff
@@ -57,6 +57,11 @@ struct NucleoF429ZI {
         VirtualMuxAlarm<'static, stm32f4xx::tim2::Tim2<'static>>,
     >,
     gpio: &'static capsules::gpio::GPIO<'static>,
+    /* uncomment to use an HD47880 LCD
+    lcd: &'static capsules::hd44780::HD44780<
+        'static,
+        VirtualMuxAlarm<'static, stm32f4xx::tim2::Tim2<'static>>,
+    >,*/
```
Comment (hudson-ayers) [inline boards/nucleo_f429zi/src/main.rs]: I am also inclined not to merge these commented out capsule instantiations into master, as they clutter main.rs and most people are unlikely to use this particular capsule

Comment (teonaseverin) [reply boards/nucleo_f429zi/src/main.rs]: I removed them from the main.rs file.

File: boards/nucleo_f429zi/Makefile (hunk 2 above)
Comment (teonaseverin) [reply boards/nucleo_f429zi/Makefile]: Yes, sorry. I modified the Makefile.

File: capsules/README.md (hunk 1 above)
Comment (teonaseverin) [reply capsules/README.md]: I moved it. Thank you.

File: boards/nucleo_f429zi/Makefile [hunk 4]
```diff
@@ -23,4 +23,4 @@ flash: target/$(TARGET)/release/$(PLATFORM).elf
 
 .PHONY: program
 program: target/$(TARGET)/debug/$(PLATFORM).elf
-	$(error See README.md and update this section accordingly)
+	$(error See README.md and update this section accordingly)
```
Comment (bradjc) [inline boards/nucleo_f429zi/Makefile]:
```suggestion
	$(error See README.md and update this section accordingly)

```

**Comment by alevy** [comment]
Thanks for writing this driver! It's high time we had support for some sort of display!

This is great for now, but might be worth starting us thinking about how to maybe abstract the userspace interface more so processes can be written in non-display-device-specific ways.

**Pull request merged**